src/classes/CRLP_Batch_Base.cls

```diff
@@ -326,6 +335,214 @@ public abstract class CRLP_Batch_Base {
+                    whereClause += '>= LAST_N_YEARS:' + maxYearsAgo;
+                }
+            } else if (maxDaysAgo != null) {
+                whereClause += '>= LAST_N_DAYS:' + maxDaysAgo;
+            } else if (maxYearsAgo != null) {
+                whereClause += '>= LAST_N_YEARS:' + maxYearsAgo;
+            }
+        }
+        return whereClause += ')';
+    }
+
+    /**
+     * @description The max Integer for all active time-bound Days_Back rollups,
+     * or null if there are no active time-bound Days_Back rollups for this summary object
+     */
+    private Integer maxDaysAgo {
+        get {
+            if (maxDaysAgo == null) {
+                maxDaysAgo = getMaxInteger(CRLP_Operation.TimeBoundOperationType.Days_Back);
+            }
+            return maxDaysAgo;
+        }
+        private set;
+    }
+
+    /**
+     * @description The max Integer for all active time-bound Years_Ago rollups,
+     * or null if there are no active time-bound Years_Ago rollups for this summary object
+     */
+    private Integer maxYearsAgo {
+        get {
+            if (maxYearsAgo == null) {
+                maxYearsAgo = getMaxInteger(CRLP_Operation.TimeBoundOperationType.Years_Ago);
+            }
+            return maxYearsAgo;
+        }
+        private set;
+    }
+
+    private Integer getMaxInteger(CRLP_Operation.TimeBoundOperationType timeBoundType) {
+        Integer maxInteger = null;
+        for (Rollup__mdt rollup : statefulCacheOfRollupsToBeProcessed) {
+            if (rollup.Active__c == true
+                    && String.isNotBlank(rollup.Time_Bound_Operation_Type__c)
+                    && rollup.Time_Bound_Operation_Type__c == timeBoundType.name()
+                    && rollup.Integer__c != null) {
+                if (maxInteger == null || rollup.Integer__c > maxInteger) {
+                    maxInteger = (Integer) rollup.Integer__c;
+                }
+            }
+        }
+        return maxInteger;
+    }
+
+    private Boolean isIncrementalMode {
+        get {
+            if (isIncrementalMode == null) {
+                isIncrementalMode = getIncrementalModeEnabled();
+            }
+            return isIncrementalMode;
+        }
+        private set;
+    }
+
+    /**
+     * @description Determines whether the rollup should be run in incremental mode
+     * @param jobType the type of rollup job
+     * @return true if the rollup should be run in incremental mode, otherwise false
+     */
+    private Boolean getIncrementalModeEnabled() {
+        if (isScheduledJob == null || isScheduledJob == false) {
+            return false;
+        }
+
+        // Only run incremental mode when the custom settings are configured
+        // to enable incremental mode for this job type
+        Customizable_Rollup_Settings__c customizableRollupSettings = UTIL_CustomSettingsFacade.getCustomizableRollupSettings();
+        if (jobType == CRLP_RollupProcessingOptions.RollupType.AccountHardCredit
```

Comment:
This needs to check for skew/non-skew
`jobMode == CRLP_RollupProcessingOptions.BatchJobMode.NonSkewMode`

Reply:
```
if (jobMode == CRLP_RollupProcessingOptions.BatchJobMode.NonSkewMode) {
   switch (jobJtype) {
       when (CRLP_RollupProcessingOptions.RollupType.ContactHardCredit) {
           return customizableRollupSettings.ContactHardCreditNonSkew_Incremental__c;
       }
else {
}
```